### ISSUE TITLE
New Version 1.6

### ISSUE THREAD

**Issue by nicojako** [anaisbetts/midi-mixer-wavexlr#13]
Hi after Update Wave XLR to 1.6 is the Plugin without function.

**Comment by bustoscarlos** [comment]
Same issue here. my stream deck required that I upgrade to 1.6 in order for some of the apps to continue working, but now I can't map my slider and faders.

**Comment by druyda** [comment]
Same issue.. In the end I have reinstalled the previous version(1.5), but I need to update to 1.6

**Comment by TheKJAMP** [comment]
have the same problem. with the wave 3. 
am back to 1.5, so that everything runs again.
Hope you can fix it. Thanks for your work and have a nice evening.

**Comment by anaisbetts** [comment]
Well that's annoying. I'll have a look this weekend.

**Comment by riomio1** [comment]
Same here. Hope you can update the plugin or something :)

**Comment by fu-raz** [comment]
They changed just about everything as far as I can tell. For starters: "getAllChannelInfo" no longer exists. Also, none of the methods on the WaveLinkClient return data anymore, they all trigger events. So this is going to be a lot of work I'm afraid.

**Comment by TheKJAMP** [comment]
Would it be better to stay on 1.5 for now? So that everything works as it should? Or can we update without hesitation?

**Comment by Kumi-de** [comment]
Maybe on purpose that no method returns anything. Elgato just released the StreamDeck+... so changing the communication to not working with midi peripherials might turn out handy for sales...

**Comment by bustoscarlos** [comment]
Good luck Dev team. Thanks for all the hard work and hope you can figure it out.

**Comment by anaisbetts** [comment]
Yeah I mean, they refactored a bunch of things so it's hard to see what's still available, we'll see what the wire changes are vs the code changes

**Comment by fu-raz** [comment]
> Maybe on purpose that no method returns anything. Elgato just released the StreamDeck+... so changing the communication to not working with midi peripherials might turn out handy for sales...

I'm sure they did this for the new StreamDeck, but I don't think it's on purpose to stop people from using midi peripherials. This is a very clean way to code.
I've also seen some additional changes that are positive. I believe the API now correctly handles linked stream/monitor sliders and I think it should be possible to have that functionality in the Midi mixer plugin. 

So let's hope it's not too much work to implement the new 1.6 version 🤞

**Comment by Kumi-de** [comment]
> > Maybe on purpose that no method returns anything. Elgato just released the StreamDeck+... so changing the communication to not working with midi peripherials might turn out handy for sales...
> 
> I'm sure they did this for the new StreamDeck, but I don't think it's on purpose to stop people from using midi peripherials. This is a very clean way to code. I've also seen some additional changes that are positive. I believe the API now correctly handles linked stream/monitor sliders and I think it should be possible to have that functionality in the Midi mixer plugin.
> 
> So let's hope it's not too much work to implement the new 1.6 version 🤞

It was just an idea... no beef intended.

**Comment by fu-raz** [comment]
> > > Maybe on purpose that no method returns anything. Elgato just released the StreamDeck+... so changing the communication to not working with midi peripherials might turn out handy for sales...
> > 
> > 
> > I'm sure they did this for the new StreamDeck, but I don't think it's on purpose to stop people from using midi peripherials. This is a very clean way to code. I've also seen some additional changes that are positive. I believe the API now correctly handles linked stream/monitor sliders and I think it should be possible to have that functionality in the Midi mixer plugin.
> > So let's hope it's not too much work to implement the new 1.6 version 🤞
> 
> It was just an idea... no beef intended.

Oh sorry bout that, didn't mean to be a dick

**Comment by anaisbetts** [comment]
Yeah, I also agree that it most likely was not intentional, just an effect of cleaning up a lot of honestly Questionable Code 😅

**Comment by DrMicka** [comment]
Hello,
I will follow with intention your progress on this update because your plugin is really useful for me who made a DIY streamdeck.
I wish I could help you but unfortunately I can only cross my fingers for you,
good luck

**Comment by fu-raz** [comment]
@anaisbetts I don't know if you've had time to look into this, but I have. I made like a proof-of-concept here: [midi-mixer-wavelink-new](https://github.com/fu-raz/midi-mixer-wavelink-new)

I did have to change things in all of the WaveLinkClient lib files. So there goes easy implementation when they release another update. Also couldn't be bothered by doing any of this in TS and worked in the lib folder directly. 

Working:

- Connects to WaveLinkClient
- Adds mixers for every available channel
- Can control volume for stream/local channels (bi-directional, so changes in wavelink app also reflect in midimixer)
- Mute works for stream/local
- Mixers for local and stream mix
- Reconnects on connection loss
- Creates solo buttons for all inputs
- Lists all available outputs and allows you to switch between them via MIDI Mixer

**Comment by riomio1** [comment]
any news on this one? :)

**Comment by boscfaad** [comment]
Hello there! Is there any chance to get this working at 1.6 again?

**Comment by TheKJAMP** [comment]
For now, stick with version 1.5 as it is stable and reliable. Just a tip.

**Comment by riomio1** [comment]
we’re already on 1.7.9 so focus on that right away maybe

**Comment by mattague** [comment]
where did you get 1.7.9? I only have 1.6.2. On that note, @fu-raz has a version that has worked flawlessly for me since i switched to it. haven't noticed any downsideds yet.

**Comment by boscfaad** [comment]
> where did you get 1.7.9? I only have 1.6.2. On that note, @fu-raz has a version that has worked flawlessly for me since i switched to it. haven't noticed any downsideds yet.

I´ll try this version soon, maybe it´ll work for me as well. Thx!!

**Comment by riomio1** [comment]
saw it on the hompage that it is the current version but im on 1.6.2 aswell so forget what i said.

**Comment by Baalim92** [comment]
Hi,

Is there any news about this issue ? Wavelink is not on 1.8 and still not working with the pluggin.
Being stuck with 1.5 is quite annoying because this version is fairly unstable a crashes a lot

**Comment by boscfaad** [comment]
> Hi,
> 
> 
> 
> Is there any news about this issue ? Wavelink is not on 1.8 and still not working with the pluggin.
> 
> Being stuck with 1.5 is quite annoying because this version is fairly unstable a crashes a lot

Hi,
There is no new version or fix for this issue.
I switched to the @fu-raz plugin and it works fine for any version since 1.6 (just in case you wanna bring it to work again)

**Comment by Baalim92** [comment]
> > Hi,
> > Is there any news about this issue ? Wavelink is not on 1.8 and still not working with the pluggin.
> > Being stuck with 1.5 is quite annoying because this version is fairly unstable a crashes a lot
> 
> Hi, There is no new version or fix for this issue. I switched to the @fu-raz plugin and it works fine for any version since 1.6 (just in case you wanna bring it to work again)

Thanks a lot, I just intalled it and it works perfectly
